ui/src/app/machines/components/ActionFormWrapper/CloneForm/CloneResults/CloneResults.tsx

```diff
@@ -0,0 +1,105 @@
+import { useEffect, useState } from "react";
+
+import { Button, Col, Notification, Row } from "@canonical/react-components";
+import pluralize from "pluralize";
+import { useSelector } from "react-redux";
+import { Link } from "react-router-dom";
+
+import machineURLs from "app/machines/urls";
+import machineSelectors from "app/store/machine/selectors";
+import type { Machine, MachineDetails } from "app/store/machine/types";
+import { NodeActions } from "app/store/types/node";
+
+type CloneError =
+  | string
+  | {
+      destinations: {
+        code: string;
+        message: string;
+      }[];
+    };
+
+type Props = {
+  closeForm: () => void;
+  destinations: Machine["system_id"][];
+  sourceMachine: MachineDetails | null;
+};
+
+const getResultsString = (count: number, error: CloneError | null) => {
+  let successCount: number;
+  if (!error) {
+    successCount = count;
+  } else if (typeof error === "object" && "destinations" in error) {
+    successCount = count - error.destinations.length;
+  } else {
+    // If an error is returned and it's not tied to the selected destinations,
+    // assume the error is global and therefore no machines were cloned to
+    // successfully.
+    successCount = 0;
+  }
+
+  return `${successCount} of ${pluralize(
+    "machine",
+    count,
+    true
+  )} cloned successfully from`;
+};
+
+export const CloneResults = ({
+  closeForm,
+  destinations,
+  sourceMachine,
+}: Props): JSX.Element | null => {
+  const [destinationCount, setDestinationCount] = useState(0);
+  const cloneError: CloneError | null =
+    useSelector(machineSelectors.eventErrors).find(
```

Comment:
Would `machineSelectors.eventErrorsForIds` work here instead of needing to do the `find` afterwards?

Reply:
The error is supposed to be the latest clone error by the machine. `machineSelectors.eventErrorsForIds` would get me a list of errors, but they wouldn't necessarily be clone errors and I would still need to manually get the first error in the list. We could add some extra selectors but it seems pretty specific to this component for now.

Reply:
Oh wait I completely missed that `eventErrorsForIds` already takes an `event` arg! My bad, that actually is a bit better